src/crash_screen/pages/page_about.c

```diff
@@ -0,0 +1,251 @@
+#ifdef VERSION_JP
+DEF_REGION_NAME(jp);
+#elif VERSION_US
+DEF_REGION_NAME(us);
+#elif VERSION_EU
+DEF_REGION_NAME(eu);
+#elif VERSION_SH
+DEF_REGION_NAME(sh);
+#elif BBPLAYER
+DEF_REGION_NAME(bb);
+#else
+DEF_REGION_NAME(xx);
+#endif
+
+// osTvType strings:
+static const char osTvTypeStrings[][5] = {
+    [OS_TV_PAL ] = "pal",
+    [OS_TV_NTSC] = "ntsc",
+    [OS_TV_MPAL] = "mpal",
+};
+
+// Microcode string:
+#define DEF_UCODE_NAME(name) static const char* ucode_name = TO_STRING2(name);
+#ifdef L3DEX2_ALONE
+DEF_UCODE_NAME(L3DEX2_alone);
+#elif F3DZEX_GBI_2
+DEF_UCODE_NAME(f3dzex2_PosLight);
+#elif F3DZEX_NON_GBI_2
+DEF_UCODE_NAME(f3dzex2_Non_PosLight);
+#elif F3DEX2PL_GBI
+DEF_UCODE_NAME(F3DEX2_PosLight);
+#elif F3DEX_GBI_2
+DEF_UCODE_NAME(F3DEX2);
+#elif F3DEX_GBI
+DEF_UCODE_NAME(F3DEX);
+#elif SUPER3D_GBI
+DEF_UCODE_NAME(Super3D);
+#else
+DEF_UCODE_NAME(Fast3D);
+#endif
+
+// Save type string:
+#define DEF_SAVETYPE_NAME(name) static const char savetype_name[8] = TO_STRING2(name);
+#ifdef EEP4K
+DEF_SAVETYPE_NAME(eep4k);
+#elif EEP16K
+DEF_SAVETYPE_NAME(eep16k);
+#elif SRAM
+DEF_SAVETYPE_NAME(sram);
+#else
+DEF_SAVETYPE_NAME(unknown);
+#endif
+
+// Compression type string:
+#define DEF_COMPRESSION_NAME(name) static const char compression_name[5] = TO_STRING2(name);
+#ifdef GZIP
+DEF_COMPRESSION_NAME(gzip);
+#elif RNC1
+DEF_COMPRESSION_NAME(rnc1);
+#elif RNC2
+DEF_COMPRESSION_NAME(rnc2);
+#elif YAYO
+DEF_COMPRESSION_NAME(yay0);
+#elif MIO0
+DEF_COMPRESSION_NAME(mio0);
+#elif UNCOMPRESSED
+DEF_COMPRESSION_NAME(none);
+#else
+DEF_COMPRESSION_NAME(unk);
+#endif
+
+// Emulator strings:
+static const EmulatorName sEmulatorStrings[] = {
+    { .bits = EMU_WIIVC,            .name = "Wii VC",           },
+    { .bits = EMU_PROJECT64_1_OR_2, .name = "pj64 1 or 2",      },
+    { .bits = EMU_PROJECT64_3,      .name = "pj64 3",           },
+    { .bits = EMU_PROJECT64_4,      .name = "pj64 4",           },
+    { .bits = EMU_MUPEN_OLD,        .name = "mupen (old)",      },
+    { .bits = EMU_MUPEN64PLUS_NEXT, .name = "mupen64plus-next", },
+    { .bits = EMU_CEN64,            .name = "cen64",            },
+    { .bits = EMU_SIMPLE64,         .name = "simple64",         },
+    { .bits = EMU_PARALLELN64,      .name = "ParallelN64",      },
+    { .bits = EMU_ARES,             .name = "ares",             },
+    { .bits = EMU_CONSOLE,          .name = "CONSOLE",          },
+};
+
+const char* get_emulator_name(enum Emulator emu) {
+    for (int i = 0; i < ARRAY_COUNT(sEmulatorStrings); i++) {
+        if (emu == sEmulatorStrings[i].bits) {
+            return sEmulatorStrings[i].name;
+        }
+    }
+
+    return NULL;
+}
+
+void page_about_init(void) {
+
+}
+
+const char gValNames_no_yes[][4] = {
+    [FALSE] = "NO",
+    [TRUE ] = "YES",
+};
+
+extern const u8 gRomSize[];
+
+#ifndef UNF
+    #define debug_is_initialized() FALSE
+#endif // !UNF
+
+#ifndef LPL_ABI_VERSION_CURRENT
```

Comment:
Just add an `#include "lib/libpl/libpl.h"` inside an `#ifdef LIBPL` instead of redefining macros.

Reply:
It's only redefined if it's not already defined, which only happens if LIBPL isn't defined. The header include is added now though.

Reply:
Why would you even want that to be defined if LIBPL isn't defined?

Reply:
Avoided a compiler error later on, but the current usage of it is ifdef'd anyway so it doesn't really change anything.